search/includes/classes/class-health.php

```diff
@@ -361,11 +369,31 @@ public function validate_index_posts_content( $start_post_id, $last_post_id, $ba
 
 		if ( $process_parallel_execution_lock ) {
 			$this->remove_validate_content_lock();
+			$this->remove_validate_content_process();
 		}
 
 		return $results;
 	}
 
+	/**
+	 * Method checks if there is an abandoned process stored. This should only happen when the validate_contents process exits unexpectedly.
+	 * In all other cases the process information should have been removed at the end of processing. This tool enables us
+	 * to potentially pick-up where we left of on long running validate contents that got interrupted.
+	 *
+	 * @return int|bool returns the ID of the first post in a batch that was process when process was updated OR false if no such values is saved.
+	 */
+	public function get_validate_content_abandoned_process() {
+		return get_option( self::CONTENT_VALIDATION_PROCESS_OPTION );
+	}
+
+	public function update_validate_content_process( $next_post_id ) {
+		update_option( self::CONTENT_VALIDATION_PROCESS_OPTION, $next_post_id );
```

Comment:
We'll need to pass `false` as the 3rd param to prevent this option from being autoloaded.
```suggestion
		update_option( self::CONTENT_VALIDATION_PROCESS_OPTION, $next_post_id, false );
```